README.md

```diff
@@ -4,37 +4,680 @@ Sylvia is the old name meaning Spirit of The Wood.
+    }
+}
+
+const CONTRACT: MyContract = MyContract::new();
+
+#[entry_point]
+fn instantiate(deps: DepsMut, env: Env, info: MessageInfo, msg: InstantiateMsg) -> StdResult<Response> {
+    msg.dispatch(&CONTRACT, (deps, env, info))
+}
+```
+
+We need to add this generic lifetime because of an optimization in storage plus -
+it doesn't want to take an owned string, as we often pass there a static string,
+but it also doesn't want to fix the `'static` ownership. 99% of the time, you can
+get away with passing just `'static` as the first `Item` generic argument, but I
+find it more convenient to introduce this "proxy" lifetime passing everywhere. I
+eliminate it in the `new` constructor, where I create the storage-plus accessors
+giving them proper keys.
+
+Note that I make the contract constructor const - it is possible as all my storage
+accessors are const, allowing me to create a single global contract to access entry
+points. Still, creating a contract on the entry point itself is possible - it is
+just a matter of taste.
+
+Now let's pass the initial counter state as a function argument:
+
+```rust
+#[contract]
+impl MyContract<'_> {
+    #[msg(instantiate)]
+    pub fn instantiate(&self, ctx: ExecCtx, counter: u64) -> StdResult<Response> {
+        self.counter.save(ctx.deps.storage, &counter)?;
+
+        Ok(Response::new())
+    }
+}
+```
+
+Sylvia would add the field into the instantiation message, which now becomes this:
+
+```rust
+struct InstantiateMsg {
+    counter: 64,
+}
+```
+
+What is essential - the field in the `InstantiateMsg` gets the same name as the
+function argument. Be careful using the typical Rust pattern to prefix it with
+`_` to leave it unused. Unfortunately, to properly silence the unused warning here,
+you need to add the `#[allow(unused)]` attribute before the argument or the whole
+function.
+
+Now let's add an execution message to the contract:
+
+```rust
+#[contract]
+impl MyContract<'_> {
+    #[msg(exec)]
+    pub fn increment(&self, ctx: ExecCtx) -> StdResult<Response> {
+        let counter = self.counter.load(ctx.deps.storage)?;
+        self.counter.save(ctx.deps.storage, &(counter + 1))?;
+        Ok(Response::new())
+    }
+}
+```
+
+Sylvia generated two message types from this:
+
+```rust
+enum ExecMsg {
+    Increment {}
+}
+
+enum ContractExecMsg {
+    MyContract(ExecMsg)
+}
+```
+
+The `ExecMsg` is the primary one you may use to send messages to the contract.
+The `ContractExecMsg` is only an additional abstraction layer that would matter
+later when we define traits for our contract. For now, we just need to use it in
+the entry point:
+
+```rust
+#[entry_point]
+fn execute(deps: DepsMut, env: Env, info: MessageInfo, msg: ContractExecMsg) -> StdResult<Response> {
+    msg.dispatch(&CONTRACT, (deps, env, info))
+}
+```
+
+One problem you might face now is that we use the `StdResult` for our contract,
+but we often want to define the custom error type for our contracts - hopefully,
```

Comment:
As in https://github.com/CosmWasm/book/pull/39... it's fortunately (not hopefully)